src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/Path.groovy

```diff
@@ -0,0 +1,42 @@
+package org.openkilda.functionaltests.helpers.model
+
+import groovy.transform.EqualsAndHashCode
+import org.openkilda.testing.model.topology.TopologyDefinition.Isl
+import org.openkilda.northbound.dto.v2.flows.FlowPathV2
+import org.openkilda.testing.model.topology.TopologyDefinition
+
+/* This class represent any kind of flow path and is intended to help compare/manipulate paths
+(presented as lists of nodes), received from different endpoints in the same manner.
+To add the new way to represent the path, just add the one into PathFactory.get() method
+Usage example:
+def nodesFromRerouteResponse = haFlowHelper.reroute(haFlowId).getSharedPath().getNodes()
+def nodesFromPathsResponse = haPathHelper.get(haFlowId).getSharedPath().getForward()
+assert pathFactory.get(nodesFromRerouteResponse) == pathFactory.get(nodesFromPathsResponse)
+
+This class has to replace *PathHelper in future
+ */
+@EqualsAndHashCode(excludes = "topologyDefinition")
+class Path {
+    List<FlowPathV2.PathNodeV2> nodes
+    TopologyDefinition topologyDefinition
+
+    Path(List<FlowPathV2.PathNodeV2> nodes, TopologyDefinition topologyDefinition) {
+        if (nodes.size() % 2 != 0) {
+            throw new IllegalArgumentException("Path should have even amount of nodes")
+        }
+        this.nodes = nodes
+        this.topologyDefinition = topologyDefinition
+    }
+
+    List<Isl> getInvolvedIsls() {
+        def isls = topologyDefinition.getIsls() + topologyDefinition.getIsls().collect { it.reversed }
+        nodes.collate(2).collect { List<FlowPathV2.PathNodeV2> pathNodes ->
+            isls.find {
+                it.srcSwitch.dpId == pathNodes[0].switchId &&
+                        it.srcPort == pathNodes[0].portNo &&
+                        it.dstSwitch.dpId == pathNodes[1].switchId &&
+                        it.dstPort == pathNodes[1].portNo
+            }
+        }
+    }
```

Comment:
This may potentially have issues if we 'create' new isls (by re-plugging via a-switch for example) that did not originally exist in the topologyDefinition. But it's ok for initial implementation

Reply:
Probably, it worth to add a warning for future.
There should be another method (or flag), which is 'heavy' (with requests to database) and rarely used. Opposite to this 'light' one, often used.